### PR TITLE
chore: WIP: Switch to Traefik 2 (closes #13)

### LINKED ISSUE #13: Upgrade Traefik to version 2
We're currently using Traefik 1.7.

We should upgrade to version 2 at some point in the future, but it's not immediately necessary.

### PR DESCRIPTION
**Pull request by YtvwlD**
This is a mostly 1:1 conversion of our config to Traefik 2.

(Why WIP? The log level and the ACME url.)

### CODE REVIEW

File: enabled/traefik.yml [hunk 1]
```diff
@@ -2,25 +2,27 @@ version: '3.7'
...
       - /var/run/docker.sock:/var/run/docker.sock
-      - acme:/etc/traefik/acme
+      - acme2:/etc/traefik/acme
       - tmp:/tmp
     networks:
       - traefik
     ports:
       - 80:80
-      - 443:443
+      - 443:80
```
Comment (nomaster) [inline enabled/traefik.yml:25]: Mapping Port 443 to port 80? Did I miss something or is this a typo?

File: configs/traefik-static.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  [providers.docker]
+    watch = true
+    swarmMode = true
+    # only detect services that have something configured
+    constraints = "Label(`traefik.docker.network`, `traefik_net`)"
+
+[api]
+
+[log]
+  level = "DEBUG"
```
Comment (nomaster) [inline configs/traefik-static.toml:13]: I think for productive use, the log level should be lowered to `ERROR`

File: configs/traefik-static.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+[certificatesResolvers]
+  [certificatesResolvers.default]
+    [certificatesResolvers.default.acme]
+      email = "account+letsencrypt@chaosdorf.de"
+      caServer = "https://acme-staging-v02.api.letsencrypt.org/directory"
```
Comment (nomaster) [inline configs/traefik-static.toml:19]: We of course want to use non-staging ACME server for production

File: configs/traefik-static.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+[providers]
+  [providers.file]
+    filename = "/etc/traefik/dynamic.toml"
+  [providers.docker]
+    watch = true
+    swarmMode = true
+    # only detect services that have something configured
+    constraints = "Label(`traefik.docker.network`, `traefik_net`)"
+
+[api]
```
Comment (nomaster) [inline configs/traefik-static.toml:10]: We used HTTP digest auth to access the API and dashboard before the change. I think this should be here as well.

File: enabled/traefik.yml (hunk 1 above)
Comment (YtvwlD) [reply enabled/traefik.yml:25]: No, this works. Traefik seems to be able to determine whether incoming traffic is TLS or plain.

File: configs/traefik-static.toml (hunk 4 above)
Comment (YtvwlD) [reply configs/traefik-static.toml:10]: Yes, we still do. But this is now configured [on the Traefik container](https://github.com/chaosdorf/docker-stacks/pull/17/files#diff-6e5588e2665a86aedfc267c7ba86d8ecR54).

File: configs/traefik-static.toml (hunk 2 above)
Comment (YtvwlD) [reply configs/traefik-static.toml:13]: We previously had `INFO` configured, but yes, this (and the ACME URL) should be changed before deploying this into production.

**Comment by YtvwlD** [comment]
So, I've changed the log level to `INFO` and removed the ACME staging URL.

I think this is good to go.

**Pull request merged**